### DISCUSSION TITLE
ISR does not work with getInitialProps in _app

### DISCUSSION THREAD

**Discussion by vincentboiardt** [vercel/next.js#22193] [Help]
**Update**: Removing `getInitialProps` from `_app` made ISR work.

Is this to be expected? If so, perhaps it should be added to the documentation?

**Original post**
Hi,

Is Incremental Static Generation only available for applications hosted on Vercel?

For my application running behind a Nginx reverse proxy the data never seem to invalidate/regenerate.

I haven't been able to find any information regarding this.

**Comment by jamesmosier** [comment]
Incremental Static Generation via `revalidate` works on any platform. Ultimately the logic for that is within the Next package running in your Node environment, not dependent on Vercel.

**Reply by vincentboiardt** [reply]
Thanks @jamesmosier ! That was what I was hoping for.

Is there any other circumstances that could stop ISG from working? (e g opting out of Automatic Static Optimization, dynamic routes, unamed exports)

**Reply by vincentboiardt** [reply]
I noticed that dynamic routes does not show `(ISR: XX Seconds)` on build for example

**Reply by jamesmosier** [reply]
I can't think of any specific reason why it would not work, but I do not know all the edge cases out there.

**Comment by vincentboiardt** [comment]
Update: Removing `getInitialProps` from `_app` made ISR work.

Is this to be expected? If so, perhaps it should be added to the documentation?

**Reply by KorayAydemir** [reply]
This should definitely be in documentation in my opinion. I've read about this on stackoverflow and googled my way here to learn if its true. Saved me quite a bit of headache of debugging later.

**Comment by vincentboiardt** [comment]
Updated the topic and main question to help others with similar issues
